pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/ControllerPeriodicTasksIntegrationTests.java

```diff
@@ -472,7 +474,26 @@ public void afterBrokerResourceValidationManagerTest(ITestContext context)
     dropOfflineTable(table2);
   }
 
-  // TODO: tests for other ControllerPeriodicTasks (RetentionManager, OfflineSegmentIntervalChecker, RealtimeSegmentValidationManager)
+  @Test
+  public void testOfflineSegmentIntervalChecker()
+      throws Exception {
+    OfflineSegmentIntervalChecker offlineSegmentIntervalChecker = _controllerStarter.getOfflineSegmentIntervalChecker();
+    ValidationMetrics validationMetrics = offlineSegmentIntervalChecker.getValidationMetrics();
+
+    // Wait until OfflineSegmentIntervalChecker gets executed
+    TestUtils.waitForCondition(
+        input -> validationMetrics.getValueOfGuage("pinot.controller.mytable_OFFLINE.SegmentCount") > 0, 60_000,
+        "Timed out waiting for OfflineSegmentIntervalChecker");
+
+    // Test the validation metrics values updated by OfflineSegmentIntervalChecker against the known values
+    // from segment metadata
+    Assert.assertEquals(validationMetrics.getValueOfGuage("pinot.controller.mytable_OFFLINE.SegmentCount"), 12);
```

Comment:
can we add constants for SegmentCount, missingSegmentCount, TotalDocumentCount (and maybe other such magic strings) in ValidationMetrics, and reuse those here?

Reply:
Please address this one too if possible.
Otherwise, LGTM
Thanks for adding this test!